providers/vault/vault_provider.go

```diff
@@ -0,0 +1,39 @@
+// Copyright © 2018 Banzai Cloud
+// Copyright © 2023 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vault
```

Comment:
How about we create a file provider instead first? There is no vault specific implementation in this PR and a file based one may be easier to test the secrets-init binary with.